realm/realm-library/src/main/cpp/io_realm_internal_OsObject.cpp

```diff
@@ -70,19 +70,20 @@ struct ChangeCallback {
             return;
         }
 
-        std::vector<jstring> field_names;
+        // The local ref of jstring needs to be released to void reach the local ref table size limitation.
+        std::vector<JavaGlobalRef> field_names;
         auto table = m_wrapper->m_object.row().get_table();
         for (size_t i = 0; i < change_set.columns.size(); ++i) {
             if (change_set.columns[i].empty()) {
                 continue;
             }
             // FIXME: After full integration of the OS schema, parse the column name from
             // wrapper->m_object.get_object_schema() will be faster.
-            field_names.push_back(to_jstring(env, table->get_column_name(i)));
+            field_names.push_back(JavaGlobalRef(env, to_jstring(env, table->get_column_name(i)), true));
```

Comment:
Who will delete created global references?
I think we need to call `DeleteGlobalRef` against elements in `field_names`.